kubectl-fdb/cmd/analyze.go

```diff
@@ -0,0 +1,338 @@
+		}
+
+		// Skip if the processGroup should be removed
+		// or should we check for how long they are marked as removed e.g. stuck in removal?
+		if processGroup.Remove {
+			continue
+		}
+
+		processGroupIssue = true
+		for _, condition := range processGroup.ProcessGroupConditions {
+			statement := fmt.Sprintf("ProcessGroup: %s has the following condition: %s since %s", processGroup.ProcessGroupID, condition.ProcessGroupConditionType, time.Unix(condition.Timestamp, 0).String())
+			printStatement(cmd, statement, true)
+		}
+
+		needsReplacement, _ := processGroup.NeedsReplacement(0)
+		if needsReplacement && autoFix {
+			replaceFailedInstance(&cluster, processGroup.ProcessGroupID)
+		}
+	}
+
+	if !processGroupIssue {
+		printStatement(cmd, "ProcessGroups are all in ready condition", false)
+	} else {
+		foundIssues = true
+	}
+
+	// 4. Check if all Pods are up and running
+	pods, err := getPodsForCluster(kubeClient, clusterName, namespace)
+	if err != nil {
+		return err
+	}
+
+	podIssue := false
+	var killPods []corev1.Pod
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != corev1.PodRunning {
+			podIssue = true
+			statement := fmt.Sprintf("Pod %s/%s has unexpected Phase %s with Reason: %s", namespace, pod.Name, pod.Status.Phase, pod.Status.Reason)
+			printStatement(cmd, statement, true)
+		}
+
+		replacePod := false
+		for _, container := range pod.Status.ContainerStatuses {
+			if container.Ready {
+				continue
+			}
+
+			podIssue = true
+			statement := fmt.Sprintf("Pod %s/%s has an unready container: %s", namespace, pod.Name, container.Name)
+			printStatement(cmd, statement, true)
+
+			// Replace the Pod if the container is unready for more then 30 minutes
+			if container.State.Terminated != nil && container.State.Terminated.ExitCode != 0 && container.State.Terminated.FinishedAt.Add(30*time.Minute).Before(time.Now()) {
+				replacePod = true
+			}
+		}
+
+		if replacePod && autoFix {
+			killPods = append(killPods, pod)
+		}
+
+		if pod.DeletionTimestamp != nil && pod.DeletionTimestamp.Add(5*time.Minute).Before(time.Now()) {
+			podIssue = true
+			statement := fmt.Sprintf("Pod %s/%s stuck in terminating should be deleted since %s", namespace, pod.Name, pod.DeletionTimestamp)
+			printStatement(cmd, statement, true)
+
+			// The instance should be delete so we can safely replace it
+			if autoFix {
+				replaceFailedInstance(&cluster, pod.Labels[controllers.FDBInstanceIDLabel])
+			}
+		}
+	}
+
+	if !podIssue {
+		printStatement(cmd, "Pods are all running and available", false)
+	} else {
+		foundIssues = true
+	}
+
+	// TODO: add more checks using the FDB status directly e.g. error message or something else or overloaded or matching versions
+	// We could add more auto fixes in the future for now this is fine.
+	// TODO: should we only replace/remove Pods if the cluster is fully replicated -> cluster.Status.Health.FullReplication?
+	// There shouldn't be an issue with replacing and removing a Pod.
+	if autoFix {
```

Comment:
In the long term we should try to merge these conditions into one place and use the same set for the operator and the plugin.